### ISSUE TITLE
Question about AdaLN action conditioning in ARPredictor

### ISSUE THREAD

**Issue by swsvv** [lucas-maes/le-wm#18]
Hi, thanks for your incredible work on this project!

I've been training LeWM from scratch and had a question about how actions are incorporated in the `ARPredictor`. I wanted to check whether this is an intentional design choice or if I am misinterpreting the implementation.

In `module.py`, `ConditionalBlock.forward`, `AdaLN` applies an action-conditioned shift and scale to the input:

```python                                                                                                                                                                
x = x + gate_msa * self.attn(modulate(self.norm1(x), shift_msa, scale_msa))
x = x + gate_mlp * self.mlp(modulate(self.norm2(x), shift_mlp, scale_mlp))  
```

  So the input to `self.attn` is already modulated with action-conditioned `shift_msa` and `scale_msa`.

  However, inside `Attention.__init__` there is: 

```python
self.norm = nn.LayerNorm(dim)                  
```

applied at the start of `Attention.forward`:                       

```python 
x = self.norm(x) 
```

  And inside `FeedForward`, the net `Sequential` starts with: 

```python                                                                                                                                                                  
  self.net = nn.Sequential(                                                                                                                                                                                        
      nn.LayerNorm(dim),                                                                                                                                                                                           
      nn.Linear(dim, hidden_dim),
      ...                         
  ) 
```

Because `LayerNorm` is shift- and scale-invariant (normalizing the tensor back to mean 0 and variance 1), I was wondering if this might be overwriting the shift and scale that `AdaLN` just computed. If so, it seems like the action conditioning might be erased before it reaches the actual attention matrices or MLP weights.

I wanted to check if this is the intended design, or should the internal `LayerNorm` layers in the sub-modules be removed to let the `AdaLN` conditioning flow through, following standard DiT/JEPA designs? It seems `AdaLN` is the primary mechanism to incorporate actions, so I wanted to make sure I'm not misreading the code.

Thanks for the great work!

**Comment by lucas-maes** [comment]
Hi! Thank you very much for the comment indeed there is an extra layer norm! we are aware of that you can safely remove it without impacting the performance

**Comment by swsvv** [comment]
Thanks for the quick clarification! That clears things up for me. Closing this now.

**Comment by wenshuaizhao** [comment]
Hi, 

Sorry for reopening this issue. I do not quite understand the conclusion in your discussion. Should we remove the two layer normalization or not? Or does it mean it does not affect the performance?

Thanks!

**Comment by lucas-maes** [comment]
You can yes! It does not change the observed performance! I'll close the discussion feel free to re-open if something is still unclear!

**Comment by wenshuaizhao** [comment]
Thanks for your prompt reply! If we do not remove the layernorm, does it mean the action conditioning only works via the `gate_msa ` and `gate_mlp`?
